src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -204,29 +204,43 @@ private static uint HandleEventPeerCertificateReceived(State state, ref Connecti
             SslPolicyErrors sslPolicyErrors  = SslPolicyErrors.None;
             X509Chain? chain = null;
             X509Certificate2? certificate = null;
-
-            if (!OperatingSystem.IsWindows())
-            {
-                // TODO fix validation with OpenSSL
-                return MsQuicStatusCodes.Success;
-            }
+            X509Certificate2Collection? additionalCertificates = null;
 
             MsQuicConnection? connection = state.Connection;
             if (connection == null)
             {
                 return MsQuicStatusCodes.InvalidState;
             }
 
-            if (connectionEvent.Data.PeerCertificateReceived.PlatformCertificateHandle != IntPtr.Zero)
-            {
-                certificate = new X509Certificate2(connectionEvent.Data.PeerCertificateReceived.PlatformCertificateHandle);
-            }
-
             try
             {
+                if (OperatingSystem.IsWindows())
+                {
+                    if (connectionEvent.Data.PeerCertificateReceived.PlatformCertificateHandle != IntPtr.Zero)
+                    {
+                        certificate = new X509Certificate2(connectionEvent.Data.PeerCertificateReceived.PlatformCertificateHandle);
+                    }
+                }
+                else if (connectionEvent.Data.PeerCertificateReceived.PlatformCertificateChainHandle != IntPtr.Zero)
```

Comment:
```suggestion
                else
                {
                    if (connectionEvent.Data.PeerCertificateReceived.PlatformCertificateChainHandle != IntPtr.Zero)
```

Reply:
I end up flipping the conditions as `PlatformCertificateChainHandle` is common to all platforms. Hppefully that will make it more readable.